src/backend/tests/unit/components/models/test_ChatOllama_component.py

```diff
@@ -0,0 +1,131 @@
+
+    # Assert that the ValueError is raised when an exception occurs
+    with pytest.raises(ValueError, match="Could not retrieve models"):
+        component.get_model(url)
+
+
+def test_update_build_config_mirostat_disabled(component):
+    build_config = {
+        "mirostat_eta": {"advanced": False, "value": 0.1},
+        "mirostat_tau": {"advanced": False, "value": 5},
+    }
+    field_value = "Disabled"
+    field_name = "mirostat"
+
+    updated_config = component.update_build_config(build_config, field_value, field_name)
+
+    assert updated_config["mirostat_eta"]["advanced"] is True
+    assert updated_config["mirostat_tau"]["advanced"] is True
+    assert updated_config["mirostat_eta"]["value"] is None
+    assert updated_config["mirostat_tau"]["value"] is None
+
+
+def test_update_build_config_mirostat_enabled(component):
+    build_config = {
+        "mirostat_eta": {"advanced": False, "value": None},
+        "mirostat_tau": {"advanced": False, "value": None},
+    }
+    field_value = "Mirostat 2.0"
+    field_name = "mirostat"
+
+    updated_config = component.update_build_config(build_config, field_value, field_name)
+
+    assert updated_config["mirostat_eta"]["advanced"] is False
+    assert updated_config["mirostat_tau"]["advanced"] is False
+    assert updated_config["mirostat_eta"]["value"] == 0.2
+    assert updated_config["mirostat_tau"]["value"] == 10
+
+
+@patch("httpx.Client.get")
+def test_update_build_config_model_name(mock_get, component):
+    # Mock the response for the HTTP GET request
+    mock_response = MagicMock()
+    mock_response.json.return_value = {"models": [{"name": "model1"}, {"name": "model2"}]}
+    mock_response.raise_for_status.return_value = None
+    mock_get.return_value = mock_response
+
+    build_config = {
+        "base_url": {"load_from_db": False, "value": None},
+        "model_name": {"options": []},
+    }
+    field_value = None
+    field_name = "model_name"
+
+    updated_config = component.update_build_config(build_config, field_value, field_name)
+
+    assert updated_config["model_name"]["options"] == ["model1", "model2"]
+
+
+def test_update_build_config_keep_alive(component):
+    build_config = {"keep_alive": {"value": None, "advanced": False}}
+    field_value = "Keep"
+    field_name = "keep_alive_flag"
+
+    updated_config = component.update_build_config(build_config, field_value, field_name)
+    assert updated_config["keep_alive"]["value"] == "-1"
+    assert updated_config["keep_alive"]["advanced"] is True
+
+    field_value = "Immediately"
+    updated_config = component.update_build_config(build_config, field_value, field_name)
+    assert updated_config["keep_alive"]["value"] == "0"
+    assert updated_config["keep_alive"]["advanced"] is True
+
+
+@patch(
+    "langchain_community.chat_models.ChatOllama",
+    return_value=ChatOllama(base_url="http://localhost:11434", model="llama3.1"),
+)
+def test_build_model(mock_chat_ollama, component):
```

Comment:
These are good tests, but likely fall under "integration" rather than "unit" tests, as they require integration with a third party (ollama) and additional setup / connection (llama on docker, I assume?)

Reply:
Also, we'll need a way to run these in CI, so likely will have to set up the appropriate github action to do so. Not in this PR, but in a follow up

Reply:
I initially thought the same, but with patching, we’re testing the ChatOllamaComponent in isolation, confirming it correctly initializes without requiring an actual Ollama deployment. The tests worked as planned in pytest without needing Ollama.

However, I agree that test_build_model_failure leans more toward integration since it’s tricky to simulate the failure scenario without the actual service.

For unit tests, we can keep the focus on initialization logic. For integration, we could set up tests that interact with a live Ollama instance, potentially in a follow-up PR with CI support.

How do you suggest we proceed ? Should I change/ remove these test?

Reply:
Oh yeah, no this is very good. Keep as is